### PR TITLE
Decorator versioning with split documentation

### PR DESCRIPTION
**Pull request by Firtzberg**
Split documentation as promised in https://github.com/neighborhoods/KojoWorkerDecoratorComponent/pull/14#issuecomment-824743420.
While writing the documentation I realized that 'WorkerV1' doesn't make that much sense. I renamed it to 'WorkerDecorationV1' and therefore 'WorkerV1Decorators' to 'WorkerDecorationV1Decorators'.
The upgrade guide will probably change a lot.

### CODE REVIEW

**Comment by Firtzberg** [comment]
The directory name changes are part of the first commit. Have a look at the second commit only to see the splitting of the documentation.

File: docs/UpgradeGuide.md [hunk 1]
```diff
@@ -1,53 +1,27 @@
...
+
+To quickly switch to the new file structure run the following regex inside your source and possibly test folders.
+``` bash
+grep -RiIl 'KojoWorkerDecoratorComponent' | xargs sed -i 's/Worker\\([a-zA-Z]*)Decorator/WorkerDecorationV1Decorators\\$1V1\\$1Decorator/g'
+grep -RiIl 'KojoWorkerDecoratorComponent' | xargs sed -i 's/KojoWorkerDecoratorComponent\\Worker(?!D)/KojoWorkerDecoratorComponent\\WorkerDecorationV1\\Worker/g'
+grep -RiIl 'KojoWorkerDecoratorComponent' | xargs sed -i 's/KojoWorkerDecoratorComponent\\Connection/KojoWorkerDecoratorComponent\\WorkerDecorationV1\\Connection/g'
+
+# fix getters and setters from fabricated aware traits
+grep -RiIl 'KojoWorkerDecoratorComponent' | xargs sed -i 's/etWorker([a-zA-Z]*)Decorator/etWorkerDecorationV1Decorators$1V1$1Decorator/g'
+grep -RiIl 'KojoWorkerDecoratorComponent' | xargs sed -i 's/etWorker(?!D)/etWorkerDecorationV1Worker/g'
```
Comment (jpmarcotte) [inline docs/UpgradeGuide.md:19]: 💪🏻  Nice work on making it super easy!

File: src/WorkerDecorationV1/README.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+      - [addDecoratorBuilderFactory, ['@Neighborhoods\KojoWorkerDecoratorComponent\WorkerDecorationV1Decorators\CrashedThresholdV1\CrashedThresholdDecorator\Builder\FactoryInterface']]
+      - [addDecoratorBuilderFactory, ['@Neighborhoods\KojoWorkerDecoratorComponent\WorkerDecorationV1Decorators\UserlandPdoV1\UserlandPdoDecorator\Builder\FactoryInterface']]
+      - [addDecoratorBuilderFactory, ['@Neighborhoods\KojoWorkerDecoratorComponent\WorkerDecorationV1Decorators\ExceptionHandlingV1\ExceptionHandlingDecorator\Builder\FactoryInterface']]
+
+```
+
+Decorator parameters have default values when possible.  
+The [Userland PDO Decorator](../WorkerDecorationV1Decorators/UserlandPdoV1/README.md#prefab5-connection-repository) builder requires the `Vendor\Product\Prefab5\Doctrine\DBAL\Connection\Decorator\RepositoryInterface` service, which doesn't have a default value. How to define it is shown in an [example](https://github.com/neighborhoods/KojoWorkerDecoratorComponentFitness/blob/main/src/SuccessWorker/Worker/UserlandPdoDecorator/Builder.service.yml).
```
Comment (jpmarcotte) [inline src/WorkerDecorationV1/README.md:126]: non-blocker: I strongly recommend using [reference style links](https://daringfireball.net/projects/markdown/syntax#link) in markdown files. This has 2 benefits IMO:
- Keeps the text cleaner when looking at the raw file
- Allows you to easily reference it multiple times, including in a `# References` section at the bottom, where you can keep all of the links defined.

The only downside, IMO, is that when reading the raw file, you don't immediately know the address of where the link points to,  but I've rarely found that to actually be a problem. Though you can also define them after the paragraph where they are introduced.

**Review by arielallon — APPROVED** [review]
reviewed synchronously

File: src/WorkerDecorationV1Decorators/CrashedThresholdV1/CrashedThresholdDecorator.service.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+parameters:
+  Neighborhoods\KojoWorkerDecoratorComponent\WorkerDecorationV1Decorators\CrashedThresholdV1\CrashedThresholdDecoratorInterface.threshold: 5
+  Neighborhoods\KojoWorkerDecoratorComponent\WorkerDecorationV1Decorators\CrashedThresholdV1\CrashedThresholdDecoratorInterface.delaySeconds: 1
```
Comment (jpmarcotte) [inline src/WorkerDecorationV1Decorators/CrashedThresholdV1/CrashedThresholdDecorator.service.yml:3]: Not really a blocker as much as a general question. Maybe this relates to a previous comment I made somewhere (though I can't seem to find it anymore, and which I might have not had enough context at the time), but how do we override these in userspace? And do they end up all being the same for every `CrashedThresholdDecorator`, or is there a way to specify them independently per stack? (e.g. I want tone job type to delay by an hour, but I want another to delay by only 5 minutes)

**Review by jpmarcotte — COMMENTED** [review]
Sorry for taking a while to get to this. Mostly looks good to me, but some suggestions that might warrant a cleanup PR if you feel like it and a general question that I haven't been able to figure out by looking at the docs / fitness stuff yet.

File: src/WorkerDecorationV1Decorators/CrashedThresholdV1/CrashedThresholdDecorator.service.yml (hunk 3 above)
Comment (jpmarcotte) [reply src/WorkerDecorationV1Decorators/CrashedThresholdV1/CrashedThresholdDecorator.service.yml:3]: Ah, here was the comment that I think I'm referring to: https://github.com/neighborhoods/KojoWorkerDecoratorComponentFitness/pull/1#issuecomment-824735474

File: docs/UpgradeGuide.md (hunk 1 above)
Comment (Firtzberg) [reply docs/UpgradeGuide.md:19]: I didn't try it, but expect this to be the most awesome upgrade guide ever!

File: src/WorkerDecorationV1Decorators/CrashedThresholdV1/CrashedThresholdDecorator.service.yml (hunk 3 above)
Comment (Firtzberg) [reply src/WorkerDecorationV1Decorators/CrashedThresholdV1/CrashedThresholdDecorator.service.yml:3]: When the container builder encounters multiple definitions of the same service or parameter the later will be used. An example is given in the [fitness](https://github.com/neighborhoods/KojoWorkerDecoratorComponentFitness/tree/main/src/DecoratorParameters/Worker). It is necessary to include the vendor path before including the product path.

This works nicely for subset containers, but in integration tests the same parameter gets overridden multiple times.

Comment (jpmarcotte) [reply src/WorkerDecorationV1Decorators/CrashedThresholdV1/CrashedThresholdDecorator.service.yml:3]: Gotcha, so in the case of an individual worker wanting separate parameters from others using the same decorator, we'd just need a different file per worker, and then the container builder would just need to know where to look?

Comment (Firtzberg) [reply src/WorkerDecorationV1Decorators/CrashedThresholdV1/CrashedThresholdDecorator.service.yml:3]: Yes

**Pull request merged**